Editor/VRChatGlobalColliderPass.cs

```diff
@@ -0,0 +1,171 @@
+{
+	[RunsOnPlatforms(WellKnownPlatforms.VRChatAvatar30)]
+	internal class VRChatGlobalColliderPass : Pass<VRChatGlobalColliderPass>
+	{
+		protected override void Execute(ndmf.BuildContext ctx)
+		{
+			var remapColliders = ctx.AvatarRootTransform.GetComponentsInChildren<ModularAvatarGlobalCollider>(true);
+			if (remapColliders.Length == 0) return;
+
+			remapColliders = remapColliders.OrderByDescending(c => c.ManualRemap).ToArray();
+
+			var LogRemapUsingFinger = new HashSet<GameObject>();
+			var logAutoRemapsIndexFinger = new HashSet<GameObject>();
+			var logAutoRemapsFailed = new HashSet<GameObject>();
+			
+			var usedColliders = new Dictionary<GlobalCollider, GameObject>();
+
+			var colliderPriority = new[]
+			{
+				GlobalCollider.FingerRingLeft,
+				GlobalCollider.FingerRingRight,
+				GlobalCollider.FingerMiddleLeft,
+				GlobalCollider.FingerMiddleRight,
+				GlobalCollider.FingerLittleLeft,
+				GlobalCollider.FingerLittleRight,
+				GlobalCollider.FingerIndexLeft,
+				GlobalCollider.FingerIndexRight
+			};
+
+			var ColliderIndex = 0;
+			foreach (var my in remapColliders)
+			{
+				ColliderIndex++;
+				GlobalCollider targetCollider;
+
+				var newColliderConfig = new ColliderConfig {
+					state = ColliderConfig.State.Custom,
+					isMirrored = false,
+					transform = my.RootTransform,
+					radius = my.Radius,
+					height = my.Height,
+					position = my.Position,
+					rotation = my.Rotation,
+				};
+
+				if (my.ManualRemap)
+				{
+					if (!ModularAvatarGlobalCollider.validVRChatColliders.Contains(my.ColliderToHijack))
+					{
+						//using a collider that's not something on the VRC Descriptor.
+						//May happen if a value is set via debug or if it's data from another platform.
+						BuildReport.Log(ErrorSeverity.Error,
+							"error.global_collider.invalid_manual_collider_vrc",
+							my.gameObject, my.ColliderToHijack.ToString());
+					}
+
+					targetCollider = my.ColliderToHijack;
+					if (!my.IgnoreOverwriteCheck && usedColliders.TryGetValue(targetCollider, out var overwrittenCollider))
+					{
+						//Collider was already used, so it will be overriten by this remap.
+						BuildReport.Log(ErrorSeverity.NonFatal,
+							"validation.global_collider.manual_collider_overwrite", 
+							overwrittenCollider.gameObject, my.gameObject);
+					}
+				}
+				else
+				{
+					targetCollider = GlobalCollider.None;
+					foreach (var collider in colliderPriority)
+					{
+						if (usedColliders.ContainsKey(collider))
+							continue;
+
+						if (collider == GlobalCollider.FingerIndexLeft || collider == GlobalCollider.FingerIndexRight)
+							logAutoRemapsIndexFinger.Add(my.gameObject);
+
+						targetCollider = collider;
+						break;
+					}
+					if (targetCollider == GlobalCollider.None)
+					{
+						logAutoRemapsFailed.Add(my.gameObject);
+						continue;
+					}
+				}
+
+				usedColliders[targetCollider] = my.gameObject;
+
+				var desc = ctx.VRChatAvatarDescriptor();
```

Comment:
We'll need to conditionally compile this file to avoid breaking the build when VRCSDK isn't present.